controller/api/config.go

```diff
@@ -12,14 +12,42 @@ type ConfigUnset struct {
 
 // Config is the structure of an app's config.
 type Config struct {
-	Owner    string                 `json:"owner,omitempty"`
-	App      string                 `json:"app,omitempty"`
-	Values   map[string]interface{} `json:"values,omitempty"`
-	Memory   map[string]interface{} `json:"memory,omitempty"`
-	CPU      map[string]interface{} `json:"cpu,omitempty"`
-	Tags     map[string]interface{} `json:"tags,omitempty"`
-	Registry map[string]interface{} `json:"registry,omitempty"`
-	Created  string                 `json:"created,omitempty"`
-	Updated  string                 `json:"updated,omitempty"`
-	UUID     string                 `json:"uuid,omitempty"`
+	Owner        string                 `json:"owner,omitempty"`
+	App          string                 `json:"app,omitempty"`
+	Values       map[string]interface{} `json:"values,omitempty"`
+	Memory       map[string]interface{} `json:"memory,omitempty"`
+	CPU          map[string]interface{} `json:"cpu,omitempty"`
+	Healthchecks map[string]interface{} `json:"healthchecks,omitempty"`
+	Tags         map[string]interface{} `json:"tags,omitempty"`
+	Registry     map[string]interface{} `json:"registry,omitempty"`
+	Created      string                 `json:"created,omitempty"`
+	Updated      string                 `json:"updated,omitempty"`
+	UUID         string                 `json:"uuid,omitempty"`
+}
+
+type KVPair struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type ExecProbe struct {
+	Command []string `json:"command"`
+}
+
+type HTTPGetProbe struct {
+	Path        string   `json:"path"`
+	Port        string   `json:"port"`
+	HTTPHeaders []KVPair `json:"httpHeaders"`
```

Comment:
done. added period seconds, success/failure threshold and httpHeaders but not host and scheme. also didn't add tcpSocket probe since we don't support TCP just yet